### PR TITLE
[CDAP-9290] DataPrep file browser implementation

### PR DESCRIPTION
**Pull request by elfenheart**
JIRA: https://issues.cask.co/browse/CDAP-9290

### CODE REVIEW

File: cdap-ui/app/cdap/components/FileBrowser/index.js [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+  }
+
+  renderRow(row) {
+    return (
+      <div
+        key={row.uniqueId}
+        className="row content-row"
+      >
+        <div className="col-xs-3 name">
+          <span
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/FileBrowser/index.js]: I would suggest we move away from this pattern. We would lose keyboard accessibility if we do this. We should change this in dataprep as well.

File: cdap-ui/app/cdap/components/FileBrowser/index.js [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+            className={classnames('type-icon fa fa-fw', {
+              'folder-icon fa-folder-o': row.directory,
+              'file-icon fa-file-o': !row.directory
+            })}
+          />
+          <span title={row.name}>{row.name}</span>
+        </div>
+        <div className="col-xs-2">
+          {row.directory ? 'Directory' : row.type}
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/FileBrowser/index.js]: i18n?

File: cdap-ui/app/cdap/components/FileBrowser/index.js [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+      });
+    });
+  }
+
+  formatResponse(contents) {
+    return contents.map((content) => {
+      content.uniqueId = shortid.generate();
+      content['last-modified'] = moment(content['last-modified']).format('MM/DD/YY HH:mm');
+      content.size = convertBytesToHumanReadable(content.size, HUMANREADABLESTORAGE_NODECIMAL);
+      content.type = content.type === 'UNKNOWN' ? '--' : content.type;
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/FileBrowser/index.js:100]: i18n?

**Review by ajainarayanan — COMMENTED** [review]
Couple of minor comments.

File: cdap-ui/app/cdap/components/FileBrowser/index.js (hunk 3 above)
Comment (elfenheart) [reply cdap-ui/app/cdap/components/FileBrowser/index.js:100]: It's just a dash, i'm not sure we should put that in i18n

Comment (ajainarayanan) [reply cdap-ui/app/cdap/components/FileBrowser/index.js:100]: My bad. Mistook for the `UNKNOWN`

**Comment by ajainarayanan** [comment]
👍 Apart from the i18n LGTM

**Review by ajainarayanan — APPROVED** [review]
👍

**Review by bdmogal — APPROVED** [review]
Some issues such as sorting, bread crumb styling, pushing the contents of a file to data prep, etc will be handled in a later PR. The current browsing experience looks good to me.

**Comment by ajainarayanan** [comment]
@elfenheart One thing you might want to fix is if you could avoid the last extraneous `/` at the end of the URL. Right now its a strict mode where `http://localhost:11011/cdap/ns/default/file/some/path/to/folder` will go to 404 but `http://localhost:11011/cdap/ns/default/file/some/path/to/folder/` could go to the right page.

**Pull request merged**